File: frontend-web/src/services/apiService.ts
import axios from 'axios';

// TODO: Move these to a central types.ts file
export interface User {
    email: string;
    id: number;
    is_active: boolean;
}
export interface UserCreate {
    email: string;
    password: string;
}
export interface OrderItemCreate {
    product_id: number;
    quantity: number;
}
export interface OrderCreate {
    items: OrderItemCreate[];
}

const API_URL = 'http://localhost:8000';

export const register = async (user: UserCreate): Promise<User> => {
    const response = await axios.post(`${API_URL}/users/`, user);
    return response.data;
};

export const login = async (formData: FormData): Promise<{ access_token: string }> => {
    const response = await axios.post(`${API_URL}/token`, formData, {
        headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
    });
    return response.data;
};

export const getCurrentUser = async (token: string): Promise<User> => {
    const response = await axios.get(`${API_URL}/users/me/`, {
        headers: { Authorization: `Bearer ${token}` }
    });
    return response.data;
};

export const createOrder = async (order: OrderCreate, token: string): Promise<any> => {
    const response = await axios.post(`${API_URL}/orders/`, order, {
        headers: { Authorization: `Bearer ${token}` }
    });
    return response.data;
};

export const getOrders = async (token: string): Promise<any[]> => {
    const response = await axios.get(`${API_URL}/orders/`, {
        headers: { Authorization: `Bearer ${token}` }
    });
    return response.data;
};

export const createPaymentIntent = async (order_id: number, token: string): Promise<{ client_secret: string }> => {
    const response = await axios.post(`${API_URL}/create-payment-intent`, { order_id }, {
        headers: { Authorization: `Bearer ${token}` }
    });
    return response.data;
};

export const updateOrderStatus = async (order_id: number, status: string, token: string): Promise<any> => {
    const response = await axios.post(`${API_URL}/orders/${order_id}/update-status`, { status }, {
        headers: { Authorization: `Bearer ${token}` }
    });
    return response.data;
};

File: android-app/app/src/main/java/com/example/flowershop/data/api/ApiService.kt
package com.example.flowershop.data.api

import com.example.flowershop.data.models.*
import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    
    // Authentication
    @POST("token")
    @FormUrlEncoded
    suspend fun login(
        @Field("username") email: String,
        @Field("password") password: String
    ): Token
    
    @POST("users/")
    suspend fun register(@Body user: UserCreate): User
    
    @GET("users/me/")
    suspend fun getCurrentUser(): User
    
    // Products
    @GET("products/")
    suspend fun getProducts(
        @Query("skip") skip: Int = 0,
        @Query("limit") limit: Int = 100
    ): List<Product>
    
    @POST("products/")
    suspend fun createProduct(@Body product: ProductCreate): Product
    
    // Orders
    @POST("orders/")
    suspend fun createOrder(@Body order: OrderCreate): Order
    
    @GET("orders/")
    suspend fun getOrders(): List<Order>
    
    @POST("orders/{order_id}/cancel")
    suspend fun cancelOrder(@Path("order_id") orderId: Int): Order
    
    @POST("orders/{order_id}/process-payment")
    suspend fun processPayment(@Path("order_id") orderId: Int): Response<Any>
    
    // Payment
    @POST("create-payment-intent")
    suspend fun createPaymentIntent(@Body request: PaymentIntentCreateRequest): PaymentIntentResponse
}

